apps/api/src/app/shared/services/cache/invalidate-cache.service.ts

```diff
@@ -26,6 +27,40 @@ export class InvalidateCacheService {
     }
   }
 
+  public async invalidateSubscriber({ credentials }: { credentials: { _id: string; _environmentId: string } }) {
+    if (!this.cacheService?.cacheEnabled()) return;
+
+    const keyBySubscriberId = commonBuilder(credentials);
+
+    const subscriberById = await this.cacheService.get(keyBySubscriberId);
+
+    if (!subscriberById) return;
+
+    const keyBySubscriberSubscriberId = subscriberBuilder(JSON.parse(subscriberById));
+
+    try {
+      await this.cacheService.del([subscriberById, keyBySubscriberSubscriberId]);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(
+        `An error has occurred when deleting "key: ${subscriberById}, ${keyBySubscriberSubscriberId}",`,
+        'InvalidateCache',
+        err
+      );
+    }
+  }
```

Comment:
So the extra logic here is the following whenever we invalidate we have the subscriber._id therefore we can fetch it from the Redis and invalidate by both (subscriber._id 
 && subscriber.subscriberId ) 

There is still a possibility that in Redis will be a record of subscriber under subscriber.subscriberId and not by subscriber._id , that mean that subscriber.subscriberId  wont be validated. (it will once a record will be created / TTL).

Reply:
Removed this one.